src/spark/utils.py

```diff
@@ -10,6 +10,7 @@
 HADOOP_AWS_VER = os.getenv('HADOOP_AWS_VER')
 DELTA_SPARK_VER = os.getenv('DELTA_SPARK_VER')
 SCALA_VER = os.getenv('SCALA_VER')
+DEFAULT_EXECUTOR_CORES = 1  # the default number of CPU cores that each Spark executor will use
```

Comment:
Without this argument how many cores does each executor use? Can you document that here?

Reply:
added comments. I noticed Spark will take all cores on the worker if we don't set this.